src/Config/CoreConfigProvider.php

```diff
@@ -0,0 +1,510 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Config;
+
+use Glpi\Dashboard\Dashboard;
+use Glpi\Dashboard\Grid;
+use Glpi\Config\Option as Option;
+
+final class CoreConfigProvider implements ConfigProviderInterface
+{
+    private static self $instance;
+
+    public static function getInstance(): self
+    {
+        if (!isset(self::$instance)) {
+            self::$instance = new self();
+        }
+
+        return self::$instance;
+    }
+
+    public function getConfigSections(): array
```

Comment:
IMHO, the `CoreConfigProvider` and the `ConfigManager` should not manage any sections. These sections are only intended to distribute configuration entries between different pages, but, because the goal is to remove advanced configuration options from the basic configuration sections, many options will ultimately have no sections at all.

The `/templates/pages/setup/general/general_setup.html.twig` could directly fetch its options one by one from the manager, instead of fetching them from an array based on a section.
```diff
- {{ _self.render_option(options['url_base'], config_scope, field_options) }}
+ {{ _self.render_option(config_manager.getOption('core', 'url_base'), config_scope, field_options) }}
```
The code of `CoreConfigProvider` and `ConfigManager` classes would also be simplified if we remove the section handling from there.

The `ConfigManager::getOption()` `context` param could probably be made optional to simplify its usage.

Reply:
> because the goal is to remove advanced configuration options from the basic configuration sections, many options will ultimately have no sections at all.

The goal was to remove advanced/uncommon options only. Maybe my idea of what is considered advanced/uncommon is different, but I think the vast majority of options will remain in the regular entity/setup UI as well.